Add 1.12 tests

wonjoolee95 opened the pull request:
Add 1.12 tests

JackCaoG commented:
I am assuming this is just copy from 1.11 and change it to 1.12? Is there a reason that it is still a draft? Do we need to dry run this before merging this?

wonjoolee95 commented:
> I am assuming this is just copy from 1.11 and change it to 1.12? Is there a reason that it is still a draft? Do we need to dry run this before merging this?

Yes, this is just a copy from 1.11 to 1.12 with some changes to 1.12 branch. We should be good to merge this, this won't affect any of the existing tests.

JackCaoG reviewed (approved):
Thanks! Let's try to get some run this weekend.

will-cromar reviewed (changes requested):
I would copy from nightly to 1.12 instead of 1.11 to 1.12. 1.11 had quite a few broken test configs that have been fixed recently. Just remove the configs that shouldn't block the release (especially PjRt).

wonjoolee95 commented:
> I would copy from nightly to 1.12 instead of 1.11 to 1.12. 1.11 had quite a few broken test configs that have been fixed recently. Just remove the configs that shouldn't block the release (especially PjRt).

Thanks for pointing that out, @will-cromar. I updated the PR to copy from nightly tests instead. Also removed tests for `pjrt`. I'll follow-up with separate PRs on removing tests that we don't need for r1.12 release.

will-cromar commented on tests/pytorch/r1.12/common.libsonnet:
```diff
@@ -0,0 +1,66 @@
+      tpuVmPytorchSetup: |||
+        sudo pip3 uninstall --yes torch torch_xla torchvision numpy
+        sudo pip3 install https://storage.googleapis.com/tpu-pytorch/wheels/tpuvm/torch-1.12-cp38-cp38-linux_x86_64.whl
+        sudo pip3 install https://storage.googleapis.com/tpu-pytorch/wheels/tpuvm/torch_xla-1.12-cp38-cp38-linux_x86_64.whl
```
Do we need to install libtpu too?

wonjoolee95 replied:
Absolutely, good catch. Added, `sudo pip3 install https://storage.googleapis.com/cloud-tpu-tpuvm-artifacts/wheels/libtpu-nightly/libtpu_nightly-0.1.dev20220518-py3-none-any.whl`.

will-cromar commented on tests/pytorch/r1.12/common.libsonnet:
```diff
@@ -0,0 +1,66 @@
+        # pytorch/xla anyway
+        mkdir pytorch
+        cd pytorch
+        git clone https://github.com/pytorch/xla.git -b r1.11
```
1.12

will-cromar reviewed (approved):
Thanks @wonjoolee95! Just a couple of comments

wonjoolee95 commented:
Thanks! I don't seem to have merging rights. Please merge this when you get the time, thanks 😄, @will-cromar!

will-cromar reviewed (commented):
I just checked the diff and there aren't any new files generated. You'll need to add 1.12 here: https://github.com/GoogleCloudPlatform/ml-testing-accelerators/blob/master/tests/pytorch/targets.jsonnet

will-cromar commented on tests/pytorch/targets.jsonnet:
```diff
@@ -15,9 +15,11 @@
 local nightly = import 'nightly/targets.jsonnet';
 local r1_10 = import 'r1.10/targets.jsonnet';
 local r1_11 = import 'r1.11/targets.jsonnet';
+local r1_11 = import 'r1.12/targets.jsonnet';
```
r1_12

wonjoolee95 replied:
Thanks lol, definitely not used to editing in vim
